### PR TITLE
[air/docs] update examples to remove pandas again

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Check if `s3` reads work; if not then revert to https

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by pcmoritz** [comment]
This doesn't work, my analysis why is here: https://github.com/ray-project/ray/issues/19799#issuecomment-1185035602

Three options:
- You use an anonymous user like I do in https://github.com/ray-project/ray/pull/26482 with `s3://anonymous@air-example-data/ocr_tiny_dataset`
- We allow S3 in the CI (probably the best solution)
- Use https

**Comment by pcmoritz** [comment]
@thomasdesr Can you allow S3 access in our CI role?

File: doc/source/ray-air/doc_code/checkpoint_usage.py [hunk 1]
```diff
@@ -23,18 +23,12 @@ def build_model() -> tf.keras.Model:
...
 from ray.train.xgboost import XGBoostTrainer
 
 
-bc_df = pd.read_csv(
-    "https://air-example-data.s3.us-east-2.amazonaws.com/breast_cancer.csv"
-)
-dataset = ray.data.from_pandas(bc_df)
-# Optionally, read directly from s3
-# dataset = ray.data.read_csv("s3://air-example-data/breast_cancer.csv")
+dataset = ray.data.read_csv("s3://air-example-data/breast_cancer.csv")
```
Comment (pcmoritz) [inline doc/source/ray-air/doc_code/checkpoint_usage.py:31]: This still needs fixing -- the others seem to be working now :)

Comment (richardliaw) [reply doc/source/ray-air/doc_code/checkpoint_usage.py]: addressed! and more!

File: doc/source/ray-air/examples/sklearn_example.ipynb [hunk 2]
```diff
@@ -82,13 +82,9 @@
    "outputs": [],
    "source": [
     "def prepare_data() -> Tuple[Dataset, Dataset, Dataset]:\n",
-    "    import pandas as pd\n",
-    "    df = pd.read_csv(\"https://air-example-data.s3.us-east-2.amazonaws.com/breast_cancer_with_categorical.csv\")\n",
-    "    dataset = ray.data.from_pandas(df)\n",
-    "    # Optionally, read directly from s3\n",
-    "    # dataset = ray.data.read_csv(\"s3://air-example-data/breast_cancer_with_categorical.csv\")\n",
+    "    dataset = ray.data.read_csv(\"s3://anonymous@air-example-data/breast_cancer.csv\")\n",
```
Comment (richardliaw) [inline doc/source/ray-air/examples/sklearn_example.ipynb:85]:
```suggestion
    "    dataset = ray.data.read_csv(\"s3://anonymous@air-example-data/breast_cancer_with_categorical.csv\")\n",
```

**Review by pcmoritz — APPROVED** [review]
Glad to see this! Need to do DCO before merging.

**Pull request merged**